pkg/querier/tripperware/query.go

```diff
@@ -721,3 +732,33 @@ func marshalHistogramBucket(b HistogramBucket, stream *jsoniter.Stream) {
 	jsonutil.MarshalFloat(b.Count, stream)
 	stream.WriteArrayEnd()
 }
+
+func (s *PrometheusResponseStats) MarshalJSON() ([]byte, error) {
+	stats := struct {
+		Samples *PrometheusResponseSamplesStats `json:"samples"`
+	}{
+		Samples: s.Samples,
+	}
+	if s.Samples.TotalQueryableSamplesPerStep == nil {
+		s.Samples.TotalQueryableSamplesPerStep = []*PrometheusResponseQueryableSamplesStatsPerStep{}
+	}
+	return json.Marshal(stats)
+}
+
+func SetRequestHeaders(h http.Header, defaultCodec string, compression string) {
+	if compression == GzipCompression {
+		h.Set("Accept-Encoding", GzipCompression)
+	}
+	if defaultCodec == "protobuf" {
+		h.Set("Accept", ApplicationProtobuf)
+	}
+	h.Set("Accept", ApplicationJson)
```

Comment:
I think the behavior of `h.Set` is that it overwrites the key with the specified value. So if default codec is `protobuf` it is still overwrite to `json` in this case.

Can we just set both protobuf and json for `protobuf` case?

Reply:
done